src/libraries/System.Memory/tests/ReadOnlySpan/GetEnumerator.cs

```diff
@@ -48,13 +51,83 @@ public static void GetEnumerator_Manual_AllValuesReturnedCorrectly(int[] array)
             Assert.False(e.MoveNext());
 
             Assert.Equal(Enumerable.Sum(array), sum);
+            Assert.Equal(Enumerable.Sum(array), SumGI(span.GetEnumerator()));
+            Assert.Equal(Enumerable.Sum(array), SumI(span.GetEnumerator()));
         }
 
         [Fact]
         public static void GetEnumerator_MoveNextOnDefault_ReturnsFalse()
         {
             Assert.False(default(ReadOnlySpan<int>.Enumerator).MoveNext());
             Assert.ThrowsAny<Exception>(() => default(ReadOnlySpan<int>.Enumerator).Current);
+
+            TestGI(default(ReadOnlySpan<int>.Enumerator));
+            TestI(default(ReadOnlySpan<int>.Enumerator));
+
+            static void TestGI<TEnumerator>(TEnumerator enumerator) where TEnumerator : IEnumerator<int>, allows ref struct
+            {
+                Assert.False(enumerator.MoveNext());
+                try { _ = enumerator.Current; } catch (Exception) { }
```

Comment:
What are all of these try/catches trying to validate? Is it trying to test that an exception gets thrown? If so, there should be asserts validating that.